api/src/middleware/authorization.js

```diff
@@ -0,0 +1,69 @@
+const auth = require('../auth'),
+      serverUtils = require('../server-utils');
+
+module.exports.ONLINE_ONLY_ENDPOINTS = [
+  '_design/*/_list/*',
+  '_design/*/_show/*',
+  '_design/*/_view/*',
+  '_find',
+  '_find/*',
+  '_explain',
+  '_explain/*',
+  '_index',
+  '_index/*'
+];
```

Comment:
All other CouchDB endpoints like `_compact`, `_view_cleanup`, `_revs_limit` authorization is already controlled by CouchDB.